Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/FirebaseCloudMessaging/FirebaseCloudMessagingConnector.cs

```diff
@@ -64,6 +72,25 @@ public void ValidateRegistration(PnsRegistration registration)
             throw new InfrastructureException(InfrastructureErrors.InvalidPushNotificationConfiguration(_options.GetSupportedAppIds()));
     }
 
+    private async Task HandleResponse(BatchResponse batchResponse, IReadOnlyList<DeviceId> devices)
+    {
+        var devicesToDelete = new List<DeviceId>();
+        for (var index = 0; index < batchResponse.Responses.Count; index++)
+        {
+            var response = batchResponse.Responses[index];
+            if (!response.IsSuccess)
+            {
+                _logger.LogError("Push notification failed for {device}: {responseMessage}", devices[index], response.Exception.Message);
```

Comment:
In case of `response.Exception.MessagingErrorCode is MessagingErrorCode.InvalidArgument or MessagingErrorCode.Unregistered`, you should not log an error. This only causes confusion when looking into the logs. You should only log with level "information" instead. But in all other cases, you can log an error.